### ISSUE TITLE
Strapi data seeding mechanism

### ISSUE THREAD

**Issue by vishi24** [beckn/strapi-plugins#70]
## Description

Create a data seeding mechanism in Strapi, facilitating seamless insertion and creation of data from CSV or XLSX files. This initiative aims to streamline the data population process within Strapi, enhancing efficiency and easing the burden on users.

#### Approach:
1. **Architectural Assessment**:
   - Evaluate the existing architecture of Strapi and identify limitations regarding data seeding functionalities.
   - Explore potential plugins or custom solutions that can extend Strapi's capabilities to support data seeding from CSV or XLSX files.

2. **Plugin Exploration**:
   - Research available plugins compatible with Strapi that offer data seeding functionalities.
   - Assess the suitability and reliability of each plugin based on factors such as community support, documentation, and compatibility with the latest versions of Strapi.

3. **Custom Solution Development**:
   - Design and develop a custom data seeding mechanism tailored to Strapi's architecture and requirements.
   - Implement features to accept CSV or XLSX files as input and process them for data insertion into Strapi.
   - Include error handling mechanisms to address scenarios such as missing domains or mandatory fields.

#### Key Considerations:
- **Domain Specific Handling**:
  - Implement a check mechanism to ensure data integrity, specifically addressing scenarios where data is mandatory for specific domains and fields are missing.
- **Scalability and Performance**:
  - Design the solution to handle large datasets efficiently while maintaining optimal performance.

## Goals

- [ ] Data seeding mechanism for Strapi.

## Expected Outcome

Fully functional data seeding mechanism in Strapi that allows data seeding for different domains.
## Acceptance Criteria

1. Implementation of a robust data seeding mechanism in Strapi, enabling easy insertion of data from CSV or XLSX files.
2. Availability of plugins or custom solutions that seamlessly integrate with Strapi, offering reliable data seeding functionalities.
3. User-friendly interface within Strapi for initiating and monitoring the data seeding process.
4. Documentation and support resources to assist users in utilizing the data seeding mechanism effectively.

**Comment by em-abee** [comment]
Hi @vishi24,

Are we aiming to seed specific data types, such as catalog data, through importing from a sheet?

**Comment by vishi24** [comment]
> Hi @vishi24,
> 
> Are we aiming to seed specific data types, such as catalog data, through importing from a sheet?

@em-abee Yes, this will be used to populate the catalog in Strapi. Basically a bulk upload of catalogs using csv file.

**Comment by em-abee** [comment]
Kool! Understood

**Comment by em-abee** [comment]
@vishi24  - I believe we'll need to develop a custom solution for seeding, as I couldn't find any existing plugins that support importing from xlsx/csv files. The closest solution I found is a plugin that supports JSON-based import. I'll attach the link to that plugin here: [https://market.strapi.io/plugins/strapi-plugin-import-export-entries](https://market.strapi.io/plugins/strapi-plugin-import-export-entries)

**Comment by iamvermajayant** [comment]
@em-abee @vishi24 Hey guys, Did you found any solution to this problem Actually I am kind of stuck in here with this issue. Let me know if u have found any other alternative to this
